Assets/Scripts/Room/Chat/ChatCommandHandler.cs

```diff
@@ -0,0 +1,777 @@
+    {
+        int playerId;
+        if (int.TryParse(playerIdString, out playerId))
+        {
+            foreach (GameObject gameObject in GameObject.FindGameObjectsWithTag("Player"))
+            {
+                if (gameObject.GetPhotonView().owner.ID == playerId)
+                {
+                    Camera.main.GetComponent<IN_GAME_MAIN_CAMERA>().setMainObject(gameObject, true, false);
+                    Camera.main.GetComponent<IN_GAME_MAIN_CAMERA>().setSpectorMode(false);
+                }
+            }
+        }
+        else
+        {
+            instance.chatRoom.OutputErrorPlayerNotFound(playerIdString);
+        }
+            
+    }
+
+    /// <summary>
+    /// Output number of titan Collisions
+    /// </summary>
+    private static void OutputCollisions()
+    {
+        int collisions = 0;
+        foreach (MindlessTitan titan in instance.getTitans())
+        {
+            if (titan.IsColliding)
+            {
+                collisions++;
+            }
+        }
+        var message = collisions.ToString();
+        instance.chatRoom.AddMessage(message);
+    }
+
+    /// <summary>
+    /// Set field of view
+    /// </summary>
+    /// <param name="input"></param>
+    private static void SetFov(string input)
+    {
+        int fov;
+        if (int.TryParse(input, out fov))
+        {
+            Camera.main.fieldOfView = fov;
+            var message = $"Field of vision set to {fov}.";
+            instance.chatRoom.AddMessage(message);
+        }
+        else
+        {
+            instance.chatRoom.OutputErrorMessage("Fov has to be a number");
+        }
+    }
+
+    /// <summary>
+    /// Enter/Exit spectate mode
+    /// </summary>
+    private static void EnterExitSpecMode()
+    {
+        string message;
+        if ((int)settings[0xf5] == 0)
+        {
+            settings[0xf5] = 1;
+            instance.EnterSpecMode(true);
+            message = FormatSystemMessage("You have entered spectator mode.");
+        }
+        else
+        {
+            settings[0xf5] = 0;
+            instance.EnterSpecMode(false);
+            message = FormatSystemMessage("You have exited spectator mode.");
+        }
+
+        instance.chatRoom.AddMessage(message);
+    }
+
+    /// <summary>
+    /// Restart current game
+    /// </summary>
+    private static void RestartGame()
+    {
+        if (isMasterClient)
+        {
+            var chatMessage = new object[] { FormatSystemMessage("MasterClient has restarted the game!"), string.Empty };
+            instance.photonView.RPC("Chat", PhotonTargets.All, chatMessage);
+            instance.restartRC();
+        }
+        else
+        {
+            instance.chatRoom.OutputErrorNotMasterClient();
+        }
+    }
+
+    /// <summary>
+    /// Send private message to player with Id
+    /// </summary>
+    /// <param name="input"></param>
+    private static void SendPrivateMessage(string[] input)
```

Comment:
This seems super confusing. `string[] input` is just the list of parameters. And we string them all back together, after they've been split up. :P
So `input` should be renamed to `parameters`. `whatIsThisVar` is the actual text pf the message to be sent. Maybe `messageText` or `rawMessage`.
Maybe we should use a `StringBuilder` here.